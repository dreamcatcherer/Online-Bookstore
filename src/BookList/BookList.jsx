import React from 'react'
import { Link } from "react-router-dom";
import PropTypes from "prop-types"
import BooksListItem from "../BookListItem/BookListItem.jsx";
import css from "./BookList.module.css";

const BooksList = props => {
  return (
    <div className={css.wholeDiv}>
      <div className={css.titleDiv}><h1>{props.pageTitle} - Next Book</h1></div>
      <ul className={css.booksList}>
        {props.books.map((book,index) => (
          <Link key={index} className={css.clickBlock} to={"/book/"+book.categoryId+"/"+book.bookId}>
            <li key={index}>
              <BooksListItem url={book.image._url} title={book.title} />
            </li>
          </Link>
        ))}
      </ul>
    </div>
  );
};

BooksList.propTypes = {
  books: PropTypes.array.isRequired
};
export default BooksList;