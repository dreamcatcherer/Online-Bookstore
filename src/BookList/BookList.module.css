.wholeDiv {
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
}

.titleDiv {
  padding-left: 22%;
  padding-right: 20%;
  color: #4d1f00;
  font-size: 15px;
  font-weight: bolder;
  font-family: 'Balsamiq Sans', cursive;
}

.booksList {
  padding-left: 15%;
  padding-right: 15%;
  list-style: none;

  display: grid;
  grid-template-columns: 1fr;
  column-gap: 30px;
  row-gap: 30px;
  grid-auto-rows: 1fr;
}

.clickBlock {
  text-decoration-line: none;
}

@media (min-width: 500px) {
  .booksList {
    grid-template-columns: 1fr;
  }
}

@media (min-width: 700px) {
  .booksList {
    grid-template-columns: 1fr 1fr;
  }
}

@media (min-width: 900px) {
  .booksList {
    grid-template-columns: 1fr 1fr 1fr;
  }
}

@media (min-width: 1200px) {
  .booksList {
    grid-template-columns: 1fr 1fr 1fr 1fr;
  }
}