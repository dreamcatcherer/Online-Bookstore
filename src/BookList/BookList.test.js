import React from "react";
import { render } from "@testing-library/react";
import BookList from "./BookList.jsx";
import { BrowserRouter } from 'react-router-dom'

it("renders the BooksList component correctly", () => {
  const test = [
    {
      categoryId: "babyTo2",
      bookId: "0001-Potty",
      title: "Potty",
      image: {
        _url: "https://images-na.ssl-images-amazon.com/images/I/9195S4FTnOL.jpg"
      },
    },
    {
      categoryId: "babyTo2",
      bookId: "0002-TouchThinkLearn-Fly",
      title: "TouchThinkLearn: Fly!",
      image: {
        _url: "https://images-na.ssl-images-amazon.com/images/I/41m71V1vZjL.jpg"
      },
    }
  ]

  // For component with <Link> must nest within <Router>
  const { container } = render(
    <BrowserRouter>
      <BookList
        books={Object.values(test)}
        pageTitle="Books for baby to 2 years old"
      />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
