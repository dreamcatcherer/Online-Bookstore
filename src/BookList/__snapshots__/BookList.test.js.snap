// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the BooksList component correctly 1`] = `
<div>
  <div
    class="wholeDiv"
  >
    <div
      class="titleDiv"
    >
      <h1>
        Books for baby to 2 years old
         - Next Book
      </h1>
    </div>
    <ul
      class="booksList"
    >
      <a
        class="clickBlock"
        href="/book/babyTo2/0001-Potty"
      >
        <li>
          <div
            class="listItem"
          >
            <div
              class="imageStyle"
            >
              <img
                alt="Potty"
                src="https://images-na.ssl-images-amazon.com/images/I/9195S4FTnOL.jpg"
              />
            </div>
            <p
              class="titleStyle"
            >
              Potty
            </p>
          </div>
        </li>
      </a>
      <a
        class="clickBlock"
        href="/book/babyTo2/0002-TouchThinkLearn-Fly"
      >
        <li>
          <div
            class="listItem"
          >
            <div
              class="imageStyle"
            >
              <img
                alt="TouchThinkLearn: Fly!"
                src="https://images-na.ssl-images-amazon.com/images/I/41m71V1vZjL.jpg"
              />
            </div>
            <p
              class="titleStyle"
            >
              TouchThinkLearn: Fly!
            </p>
          </div>
        </li>
      </a>
    </ul>
  </div>
</div>
`;
